src/python/mtapi_server.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+"""
+Implements Microsoft's MTAPI (https://docs.microsoft.com/en-us/azure/cognitive-services/translator/quickstart-translator?tabs=python).
+"""
+
+import sys
+import json
+import pymarian
+
+from flask import Flask, request
+from typing import List
+
+SOURCE_LANG = "en"
+TARGET_LANG = "de"
+
+from sacremoses import MosesPunctNormalizer
+from sentence_splitter import SentenceSplitter, split_text_into_sentences
+
+# marian = pymarian.Translator(
+#     model="model.bin", vocabs=["enu.spm", "enu.spm"],
+#     beam_search=2, normalize=1, mini_batch=1, maxi_batch=1,
+#     cpu_threads=1, output_approx_knn=[128, 1024]
+# ) ---> Translator("--model model.bin --vocabs enu.spm enu.spm --beam-search 2 --normalize 1 --mini-batch 1 --maxi-batch 1 --cpu-threads 1 --output-approx-knn 128 1024")
+
+# import argparse
+
+# parser = argparse.ArgumentParser()
+# parser.add_argument("--marian-flags", "-f")
+# parser.add_argument("--doc-model", action="store_true")
+# args = parser.parse_args()
+
+norm = MosesPunctNormalizer(lang="en")
+splitter = SentenceSplitter(SOURCE_LANG)
+
+class MarianServer:
+    def __init__(self, flags=""):
+        self.marian = None
+        self.flags = flags
+        self.reloadMarian()
+
+    def reloadMarian(self):
+        if not self.marian:
+            self.marian = pymarian.Translator("-m model/model.npz -v model/enu.deu.joint.eos.spm model/enu.deu.joint.eos.spm --mini-batch 1 --maxi-batch 1")
```

Comment:
I think these paths could be configurable via command-line options or at least extracted to a global variables.